Cookbook/Proposals/SwiftGen.md

```diff
@@ -0,0 +1,122 @@
+
+```swift
+public struct LocalizationUI {
+    public struct Common {
+        public static var ok: String { return localizedName(key: "general_ok") }
+        ...
+    }
+}
+```
+
+## Problems
+
+The current approach works, as we would expect, but there's a couple of caveats in it.
+
+#### 1. Maintenance
+
+These types at the moment are created and managed by us manually which means any new addition/change/removal needs to be done by us. This may not look very significant but considering the number of strings, more than 3k, it is a lot to be maintained/extended.
+
+#### 2. Safety
+
+While these types are useful for development by enabling autocomplete they don't provide any safety since we can add a new property for a new key without necessarily adding the new string.
+
+## Proposed Solution
+
+SwiftGen is a code generator which is able to generate all these types and properties automatically by running a script which would fix the two main problems that we have with the current approach.
+
+- Maintenance will be zero because everything will be mapped based on the localisable strings;
+- Compile-time type safety will be guaranteed because everything will be generated from the localizable strings making it impossible to use something without being in the strings file or keep using something that was deleted.
+
+SwiftGen also supports strings with dynamic input generating a function to be called with all the parameters which also ensures safety as we clearly know that a certain string requires some dynamic values.
+
+We still need to ensure this script is correctly executed every time that we add/edit/remove a string but that's an easier task to manage compared to the current approach.
+
+## Implementation
+
+Adopting SwiftGen won't be a trivial task and would require a certain level of investment from us.
+
+#### Location
+
+All the symbols that we have at the moment are defined in `BabylonDependencies` and there's no reason to change this so everything can be kept as is.
+
+#### Workflow
+
+Our current workflow consists of pulling the strings manually and then only include the ones that are part of our current work. Adopting SwiftGen does not require any changes in our workflow so it should be kept unchanged, we should pull the strings and then run the script to generate the symbols manually. This file should be committed along with the strings to the repository to ensure we test the code without any external variables, e.g. if for some reason the generation fails we will get a failure while testing the PR on CI.
+
+Considering there's two steps involved, pulling the strings and then run SwiftGen to generate the symbols, and based on the fact that apply some filtering we should discard any strings unrelated with our work before running SwiftGen otherwise we will end with symbols that will need to be discarded too.
```

Comment:
We could have the fastlane script stop for dev input, e.g.:
```
✅ Strings pulled 
⚠️ Please  discard any strings unrelated with your work and hit Enter to continue.
✅ New string constants generated!
```

Reply:
Not familiar with this in specific but sounds a great option 👍